src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -0,0 +1,153 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MagicWandService {
+  /** Return set of coordinates(formatted as a 1-D array).
+   *  Coordinates correspond to pixels considered part of the mask.
+   */
+  /* tslint:disable */
+  floodfill(imgData: ImageData, xCoord: number, yCoord: number,
+            tolerance: number): Set<number> {
+    // Use basic queue to immitate recursive approach of the stack
+    // possible TODO: set array size after mvp to optimize heap memory access
+    const visit: Array<Array<number>> = new Array();
+    const visited: Set<number> = new Set();
```

Comment:
It's a little inconsistent that you store what pixels to visit as the 2D x,y location but visited is the index of the R channel in the 1D array.  Is there a reason for this choice?

Reply:
I tried to leave it in x,y format as much as possible and convert between the two formats as little as possible in order to make passing the parameters more consistent and the code easier to look at in general. 

However, I had to omit that for the visited variable. If I had passed in an [x,y] into visited, it would have been a Set<Array<number>> instead of a Set<number> which I had trouble with because doing set operations made it compare array objects instead of the actual coordinates. I then thought it would be simplest to pass the index of R channel into visited instead of passing pairs of numbers into visited.